Tests/TuistDependenciesTests/DependenciesControllerTests.swift

```diff
@@ -208,6 +214,48 @@ final class DependenciesControllerTests: TuistUnitTestCase {
         XCTAssertFalse(dependenciesGraphController.invokedClean)
     }
 
+    func test_fetch_carthage_swiftPackageManger_throws_when_duplicatedDependency() throws {
+        // Given
+        let rootPath = try temporaryPath()
+        let dependencies = Dependencies(
+            carthage: .init([
+                .github(path: "Moya", requirement: .exact("1.1.1")),
+            ]),
+            swiftPackageManager: .init([
+                .remote(url: "Moya", requirement: .exact("2.3.4")),
+            ]),
+            platforms: [.iOS]
+        )
```

Comment:
@laxmorek Looks like here I can't use the arrays directly, probably it's the optional type that should be `ExpressibleByStringLiteral`

Reply:
Hey I added `ExpressibleByArrayLiteral` to `ProjectDescription.CarthageDependencies` and `ProjectDescription.SwiftPackageManagerDependencies`.

Models above are from `TuistCore` framework and they can be built only via `init`.